src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java

```diff
@@ -237,6 +237,22 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
       return this;
     }
 
+  /**
+   * For a given server URI, validate according to RFC 2396
+   *
+   * @return {@code this}
+   * @throws IllegalStateException
+   */
+    public Builder validateEndpoint() throws IllegalStateException {
+        try {
+            URI.create(this.server);
```

Comment:
This validates the URI syntax (which we should probably do regardless), but that doesn't verify that the endpoint is reachable and available, which, I believe, was the whole point.